tests/unit/amo/components/TestRatingManager.js

```diff
@@ -469,4 +484,250 @@ describe(__filename, () => {
+        'ReportAbuseButton--is-expanded',
+      );
+
+      const button = screen.getByRole('button', {
+        name: 'Report this add-on for abuse',
+      });
+      const clickEvent = createEvent.click(button);
+      const preventDefaultWatcher = jest.spyOn(clickEvent, 'preventDefault');
+
+      fireEvent(button, clickEvent);
+
+      expect(preventDefaultWatcher).toHaveBeenCalled();
+
+      userEvent.type(
+        screen.getByPlaceholderText(
+          'Explain how this add-on is violating our policies.',
+        ),
+        message,
+      );
+
+      userEvent.click(
+        screen.getByRole('button', { name: 'Send abuse report' }),
+      );
+
+      expect(
+        screen.getByRole('button', { name: 'Sending abuse report' }),
+      ).toHaveClass('Button--disabled');
+
+      expect(dispatch).toHaveBeenCalledWith(
+        sendAddonAbuseReport({
+          addonSlug: fakeAddon.slug,
+          errorHandlerId: 'ReportAbuseButton',
+          message,
+        }),
+      );
+
+      userEvent.click(screen.getByRole('button', { name: 'Dismiss' }));
+
+      // Dismiss should have been ignored.
+      expect(screen.getByClassName('ReportAbuseButton')).toHaveClass(
+        'ReportAbuseButton--is-expanded',
+      );
+    });
+
+    it('hides the form when Dismiss is clicked', () => {
+      render();
+
+      userEvent.click(
+        screen.getByRole('button', { name: 'Report this add-on for abuse' }),
+      );
+
+      expect(screen.getByClassName('ReportAbuseButton')).toHaveClass(
+        'ReportAbuseButton--is-expanded',
+      );
+
+      userEvent.click(screen.getByRole('button', { name: 'Dismiss' }));
+
+      expect(screen.getByClassName('ReportAbuseButton')).not.toHaveClass(
+        'ReportAbuseButton--is-expanded',
+      );
+    });
+
+    it.each([ADDON_TYPE_EXTENSION, ADDON_TYPE_STATIC_THEME])(
+      'initiates an abuse report via Firefox when the "report" button is clicked if supported and add-on type is %s',
+      (addonType) => {
+        hasAbuseReportPanelEnabled.mockImplementation(() => {
+          return true;
+        });
+        const addon = createInternalAddonWithLang({
+          ...fakeAddon,
+          type: addonType,
+        });
+        const dispatch = jest.spyOn(store, 'dispatch');
+        render({ addon });
+
+        const button = screen.getByRole('button', {
+          name: 'Report this add-on for abuse',
+        });
+        const clickEvent = createEvent.click(button);
+        const preventDefaultWatcher = jest.spyOn(clickEvent, 'preventDefault');
+
+        fireEvent(button, clickEvent);
+        expect(preventDefaultWatcher).toHaveBeenCalled();
+
+        expect(screen.getByClassName('ReportAbuseButton')).not.toHaveClass(
+          'ReportAbuseButton--is-expanded',
+        );
+        expect(dispatch).toHaveBeenCalledWith(
+          initiateAddonAbuseReportViaFirefox({ addon }),
+        );
+      },
+    );
+
+    it.each([ADDON_TYPE_DICT, ADDON_TYPE_LANG])(
```

Comment:
interesting.  We don't allow abuse reports for dictionaries?  I wasn't aware.

Reply:
See https://github.com/mozilla/addons-frontend/pull/11443/files#diff-dff643b10dc60f142319afb3ed4107d6720017a0e0ea68f5a38b705e87efad0fR90-R94.

I'm not sure if that is something that is ever planned to be implemented.